backend/fms_core/viewsets/full_sample.py

```diff
@@ -0,0 +1,235 @@
+                    if g.strip()
+                ])
+
+            derived_sample = DerivedSample.objects.create(**derived_sample_data)
+
+            sample_data = dict(
+                name=full_sample['name'],
+                volume=full_sample['volume'],
+                creation_date=full_sample['creation_date'],
+                container_id=full_sample['container'],
+                **(dict(comment=full_sample['comment']) if full_sample['comment'] is not None else dict()),
+                **(dict(coordinates=full_sample['coordinates']) if full_sample['coordinates'] is not None else dict()),
+                **(dict(concentration=full_sample['concentration']) if full_sample['concentration'] is not None else dict()),
+            )
+
+            sample = Sample.objects.create(**sample_data)
+
+            DerivedBySample.objects.create(derived_sample_id=derived_sample.id,
+                                           sample_id=sample.id,
+                                           volume_ratio=1)
+
+        except ValidationError as err:
+            raise ValidationError(err)
+
+        #Serialize created sample
+        serializer = SampleSerializer(sample)
+        sample_data = serializer.data
+
+        #Add Biosample and Derived Sample fields
+        extra_fields = dict(
+            alias=biosample.alias,
+            collection_site=biosample.collection_site,
+            individual=biosample.individual_id,
+            sample_kind=derived_sample.sample_kind_id,
+            tissue_source=derived_sample.tissue_source,
+            experimental_group=derived_sample.experimental_group,
+        )
+
+        full_data = {**sample_data, **extra_fields}
+
+        return Response(full_data)
+
+    def update(self, request, *args, **kwargs):
+        full_sample = request.data
+
+        sample_data = dict(
+            name=full_sample['name'],
+            volume=full_sample['volume'],
+            creation_date=full_sample['creation_date'],
+            container_id=full_sample['container'],
+            **(dict(comment=full_sample['comment']) if full_sample['comment'] is not None else dict()),
+            **(dict(coordinates=full_sample['coordinates']) if full_sample['coordinates'] is not None else dict()),
+            **(dict(concentration=full_sample['concentration']) if full_sample[
+                                                                       'concentration'] is not None else dict()),
+        )
+
+        #Retrive the sample to update
+        try:
+            Sample.objects.filter(pk=full_sample['id']).update(**sample_data)
+            sample_to_update = Sample.objects.get(pk=full_sample['id'])
+        except Exception as err:
+            raise ValidationError(err)
+
+        #Save the new sample
+        try:
+            sample_to_update.save()
+        except Exception as err:
+            raise ValidationError(err)
+
+        #Return new sample
+        serializer = SampleSerializer(sample_to_update)
```

Comment:
return fullsample using sample_id and derived_id to identify uniquely your full sample.

Reply:
The sample_id is returned simply as id and derived_id is returned as an array of ids derived_ids. Do we need to return them differently?

Reply:
I mean the object you return. You are right that there might be multiple derived sample id for a single full_sample update. I just would not want a sample stored in the frontend following an update. The fullsample need to be returned. When you submit your request you have the required info to get the fullsample out.